packages/react-components/src/components/TextFieldWithMention/mentionTagHelpers.ts

```diff
@@ -0,0 +1,957 @@
+          plainTextRepresentation += currentOpenTag.content;
+        } else if (currentOpenTag.subTags.length > 0) {
+          // Add text after the last tag
+          const lastSubTag = currentOpenTag.subTags[currentOpenTag.subTags.length - 1];
+          const startOfRemainingText =
+            (lastSubTag.closeTagIndex ?? lastSubTag.openTagIndex) + lastSubTag.tagType.length + 3;
+          const trailingText = currentOpenTag.content.substring(startOfRemainingText);
+          plainTextRepresentation += trailingText;
+        }
+
+        currentOpenTag.plainTextEndIndex = plainTextRepresentation.length;
+        addTag(currentOpenTag, tagParseStack, tags);
+      } else {
+        const errorMessage =
+          'Unexpected close tag found. Got "' +
+          closeTagType +
+          '" but expected "' +
+          tagParseStack[tagParseStack.length - 1]?.tagType +
+          '"';
+        throw new Error(errorMessage);
+      }
+    }
+
+    // Update parsing index; move past the end of the close tag
+    parseIndex = foundHtmlTag.startIndex + foundHtmlTag.content.length;
+  } // While parseIndex < text.length loop
+
+  return { tags, plainText: plainTextRepresentation };
+};
+
+/**
+ * Parses an open HTML tag and returns the tag data.
+ * @param tag - The HTML tag to parse.
+ * @param startIndex - The starting index of the tag in the text.
+ * @returns The tag data.
+ *
+ * @private
+ */
+export const parseOpenTag = (tag: string, startIndex: number): TagData => {
+  const tagType = tag
+    .substring(1, tag.length - 1)
+    .split(' ')[0]
+    .toLowerCase()
+    .replace('/', '');
+  return {
+    tagType,
+    openTagIndex: startIndex,
+    openTagBody: tag
+  };
+};
+
+/**
+ * Finds the next HTML tag in the given text starting from the given index.
+ * @param text - The text to search for HTML tags.
+ * @param startIndex - The index to start searching for HTML tags.
+ * @returns The next HTML tag found, or undefined if no more tags are found.
+ *
+ * @private
+ */
+export const findNextHtmlTag = (text: string, startIndex: number): HtmlTag | undefined => {
+  const tagStartIndex = text.indexOf('<', startIndex);
+  if (tagStartIndex === -1) {
+    // No more tags
+    return undefined;
+  }
+  const tagEndIndex = text.indexOf('>', tagStartIndex);
+  if (tagEndIndex === -1) {
+    // No close tag
+    return undefined;
+  }
+  const tag = text.substring(tagStartIndex, tagEndIndex + 1);
+  let type: HtmlTagType = 'open';
+  if (tag[1] === '/') {
+    type = 'close';
+  } else if (tag[tag.length - 2] === '/') {
+    type = 'self-closing';
+  }
+  return {
+    content: tag,
+    startIndex: tagStartIndex,
+    type: type
+  };
+};
+
+/**
+ * Adds a tag to the tag stack and updates the parent tag's subtags.
+ * @param tag - The tag to add.
+ * @param tagStack - The tag stack.
+ * @param tags - The array of tags.
+ *
+ * @private
+ */
+export const addTag = (tag: TagData, parseStack: TagData[], tags: TagData[]): void => {
```

Comment:
Should separate props type be created in oder not to mix up `parseStack` and `tags`?

Reply:
> Should separate props type be created in order not to mix up `parseStack` and `tags`?

I feel for this one we can leave it inline. It's not used anywhere else and it's only 3 parameters.